ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/WebSocketTest.kt

```diff
@@ -299,7 +299,7 @@ class WebSocketTest : ClientLoader() {
         }
 
         test { client ->
-            assertFailsWith<WebSocketException> {
+            assertFailsWith<Exception> {
```

Comment:
It seems like the different engines can throw different exceptions and I'm unable to figure out a mechanism to handle these exceptions gracefully, the upgrade requests are correctly failing when missing the sub protocol.